### PR TITLE
Add support for creating k8s storage pools

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Add support for managing storage pools for k8s models.
This PR is the first of many which will add storage support to k8s. There's some key points:

The storage facade used a big storageAccessor interface which mingled model related and storage related apis. This has been broken up to split out the model related apis from the storage ones There's a noop caas storage implementation which will be progressed over time. Following PRs will further split up the storage interface to separate common concerns like watching volumes etc to enable caas and iaas stuff to be shared.

The k8s test suite has been refactored to break out common functionality to set up the mock client.

## QA steps

bootstrap a k8s model
juju create-storage-pool blah
juju storage-pools etc

### CODE REVIEW

File: caas/kubernetes/provider/storage_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+var _ = gc.Suite(&storageSuite{})
+
+type storageSuite struct {
+	BaseSuite
+	k8sClient kubernetes.Interface
+}
+
+func (s *storageSuite) k8sProvider(c *gc.C) storage.Provider {
+	ctrl := gomock.NewController(c)
```
Comment (howbazaar) [inline caas/kubernetes/provider/storage_test.go]: What about Finish?

Comment (howbazaar) [reply caas/kubernetes/provider/storage_test.go]: Why not take a controller?

Comment (wallyworld) [reply caas/kubernetes/provider/storage_test.go]: fixed

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**